code/iaas/ssh-common/src/main/java/io/cattle/platform/ssh/common/SslCertificateValidationUtils.java

```diff
@@ -0,0 +1,184 @@
+package io.cattle.platform.ssh.common;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.StringReader;
+import java.security.DigestInputStream;
+import java.security.GeneralSecurityException;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PublicKey;
+import java.security.SignatureException;
+import java.security.cert.CertPathBuilder;
+import java.security.cert.CertStore;
+import java.security.cert.CertificateException;
+import java.security.cert.CollectionCertStoreParameters;
+import java.security.cert.PKIXBuilderParameters;
+import java.security.cert.PKIXCertPathBuilderResult;
+import java.security.cert.TrustAnchor;
+import java.security.cert.X509CertSelector;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.codec.binary.Hex;
+import org.bouncycastle.jce.provider.X509CertificateObject;
+import org.bouncycastle.openssl.PEMReader;
+
+public class SslCertificateValidationUtils {
+
+    public static void validateCertificate(String certInput) throws Exception {
+        X509CertificateObject cert = getCertificateFromPem(certInput);
+        cert.checkValidity();
```

Comment:
I don't think we want to check the validadity of a cert.  I don't know what that API does, but we don't want to prevent somebody from uploading a cert that has expired or is yet to be active based on the date.